poker.js

```diff
@@ -45,17 +45,21 @@ const groupedHand = (hand) => hand.reduce(
 const straight = (hand) => {
   const onlyNumbers = lettersToNumber(hand);
   const sortRankArr = onlyNumbers.sort((a, b) => a - b);
-  const lastMinusFirst = sortRankArr[4] - sortRankArr[0];
-  return lastMinusFirst;
-};
-
-const highCard = (rankArr) => {
-  let highestCard = 0;
-  for (let elem of rankArr) {
-    highestCard < elem ? highestCard = Number(elem) : highestCard;
-  }
-  return highestCard;
-};
+  const truthComparison = sortRankArr.reduce(
+    (acc, cv) => {
+      if (acc - cv === -1) {
+        return cv;
+      } else return false;
+    });
+  return truthComparison;
+};
+
+const highCard = (rankArr) => rankArr.reduce(
+  (acc, rank) => {
```

Comment:
acc could be named highestCardSoFar for example or something that you may find more descriptive